docs/smt.md

```diff
@@ -0,0 +1,225 @@
+# SMT Backend
+
+The SMT backend for act is currently able to automatically prove that:
+
+- post conditions are implied by pre conditions
```

Comment:
is this working atm?

Reply:
It's a little messy, but if there is a construcor then yes. @MrChico modified the invariant checking to also check postconditions, and in this PR I insert an implicit `True` invariant to all specs with a constructor.

I would like to pull the postcondition checking out into it's own pass so we can simplify the queries and present nicer error messages to users.